infrastructure/templates/ec2-vpn-cfn.yaml

```diff
@@ -0,0 +1,150 @@
+  # and also deploy the ec2 instance with appropriate parameters
+Mappings:
+  RegionMap:
+    us-east-1:
+      "AMI": "ami-0ed9277fb7eb570c9"
+
+Resources:
+  VPC: # vpc block
+    Type: "AWS::EC2::VPC"
+    Properties:
+      CidrBlock: 192.168.0.0/16
+      EnableDnsSupport: true
+      EnableDnsHostnames: true
+      Tags:
+        - Key: Name
+          Value: "typedb-vpc-cfn"
+
+  InternetGateway: #igw block
+    Type: "AWS::EC2::InternetGateway"
+    Properties:
+      Tags:
+        - Key: Name
+          Value: igw-typedb-1
+
+  InternetGatewayAttachment: #attaches igw to vpc
+    Type: AWS::EC2::VPCGatewayAttachment
+    Properties:
+      InternetGatewayId: !Ref InternetGateway
+      VpcId: !Ref VPC
+
+  PublicSubnet1:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      AvailabilityZone: "us-east-1a"
+      CidrBlock: 192.168.0.0/20
+      MapPublicIpOnLaunch: true
+      Tags:
+        - Key: Name
+          Value: cfn-public-subnet-1
+
+  PublicSubnet2:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      AvailabilityZone: "us-east-1b"
+      CidrBlock: 192.168.16.0/20
+      MapPublicIpOnLaunch: true
+      Tags:
+        - Key: Name
+          Value: cfn-public-subnet-2
+
+  PrivateSubnet1:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      AvailabilityZone: "us-east-1a"
+      CidrBlock: 192.168.128.0/20
+      MapPublicIpOnLaunch: false
+      Tags:
+        - Key: Name
+          Value: cfn-private-subnet-1
+
+  PrivateSubnet2:
+    Type: AWS::EC2::Subnet
+    Properties:
+      VpcId: !Ref VPC
+      AvailabilityZone: "us-east-1b"
+      CidrBlock: 192.168.144.0/20
+      MapPublicIpOnLaunch: false
+      Tags:
+        - Key: Name
+          Value: cfn-private-subnet-2
+
+  PublicRouteTable:
+    Type: AWS::EC2::RouteTable
+    Properties:
+      VpcId: !Ref VPC
+      Tags:
+        - Key: Name
+          Value: cfn-route-table
+
+  DefaultPublicRoute:
+    Type: AWS::EC2::Route
+    DependsOn: InternetGatewayAttachment
+    Properties:
+      RouteTableId: !Ref PublicRouteTable
+      DestinationCidrBlock: 0.0.0.0/0
+      GatewayId: !Ref InternetGateway
+
+  PublicSubnet1RouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref PublicRouteTable
+      SubnetId: !Ref PublicSubnet1
+
+  PublicSubnet2RouteTableAssociation:
+    Type: AWS::EC2::SubnetRouteTableAssociation
+    Properties:
+      RouteTableId: !Ref PublicRouteTable
+      SubnetId: !Ref PublicSubnet2
+
+  cfnSecgroupPrivateEc2:
+    Type: "AWS::EC2::SecurityGroup"
+    Properties:
+      GroupDescription: Enable TCP connections from public subnets
+      SecurityGroupIngress:
+        - IpProtocol: tcp
+          FromPort: 1729
+          ToPort: 1729
+          CidrIp: 192.168.16.0/20
+        - IpProtocol: tcp
+          FromPort: 1729
+          ToPort: 1729
+          CidrIp: 192.168.0.0/20
+      VpcId: !Ref VPC
+
+  EC2Instance:
```

Comment:
I think this should be moved to its own file.

Reply:
which lines / blocks are you addressing specifically? the comment only highlights L#119-122

Reply:
Sorry, the EC2 Instance for typedb.